emc-metalnx-services/src/main/java/com/emc/metalnx/services/irods/UploadServiceImpl.java

```diff
@@ -64,8 +64,7 @@ public boolean upload(MultipartFile file, String targetPath, boolean computeChec
 
 		logger.info("upload()");
 
-		if (file == null || file.isEmpty() || "".equals(targetPath) || targetPath == null || "".equals(destResc)
-				|| destResc == null) {
+		if (file == null || file.isEmpty() || "".equals(targetPath) || targetPath == null) {
```

Comment:
Potentially stupid question: I see that `destResc` is being logged on L77 below... Is that going to be a problem if it's `null` in Java?

Reply:
I am not really sure how to get the logger entries to log.  I believe I did that once before but don't remember how.  The default log level is info in metalnx but these don't seem to be logging.

If it works like System.out.println() it will just print "null".  The following just prints "null".

```
class Main {
    public static void main(String[] args) {
        String s = null;
        System.out.println(s);
    }
}
```
Just in case I updated the code to the following:

```
logger.info("destResc:{}", destResc == null ? "null" : destResc);
```

Reply:
Call the logger from a place you know will be triggered and see what happens.

Reply:
I'll double check what log4j 2 does and report back.

I'd be surprised if it didn't handle null gracefully.